src/Uno.UI/UI/Xaml/UIElement.macOS.cs

```diff
@@ -121,125 +103,36 @@ internal Windows.Foundation.Point GetPosition(Point position, global::Windows.UI
-		/// <returns></returns>
-		private static CGPoint ConvertOriginPointToView(NSView view, NSView parentView)
-		{
-			var value = CGPoint.Empty;
-			var current = view;
-
-			do
-			{
-				if (current is FrameworkElement fr)
-				{
-					value.X += (nfloat)fr.AppliedFrame.X;
-					value.Y += (nfloat)fr.AppliedFrame.Y;
-				}
-				else
-				{
-					value.X += current.Frame.X;
-					value.Y += current.Frame.Y;
-				}
-
-				current = current.Superview;
-
-			} while (current != null && current != parentView);
-
-			return value;
-		}
-
-#region DoubleTapped event
-		private void RegisterDoubleTapped(DoubleTappedEventHandler handler)
-		{
-			LogRegisterPointerCanceledNotImplemented();
-		}
-
-		private void UnregisterDoubleTapped(DoubleTappedEventHandler handler)
-		{
-			LogRegisterPointerCanceledNotImplemented();
-		}
-#endregion
-
-#region PointerCanceled event
-		private void RegisterPointerCanceled(PointerEventHandler handler)
-		{
-			LogRegisterPointerCanceledNotImplemented();
-		}
-
-		private void UnregisterPointerCanceled(PointerEventHandler handler)
-		{
-			LogUnregisterPointerCanceledNotImplemented();
-		}
-#endregion
-
-#region PointerExited event
-		private void RegisterPointerExited(PointerEventHandler handler)
-		{
-			LogRegisterPointerExitedNotImplemented();
-		}
+#if DEBUG
+		public string ShowLocalVisualTree(int fromHeight) => AppKit.UIViewExtensions.ShowLocalVisualTree(this, fromHeight);
+#endif
 
-		private void UnregisterPointerExited(PointerEventHandler handler)
-		{
-			LogUnRegisterPointerExitedNotImplemented();
-		}
-#endregion
+		/// <inheritdoc />
+		public override bool AcceptsFirstResponder() 
+			=> true; // This is required to receive the KeyDown / KeyUp. Note: Key events are then bubble in managed.
 
-#region PointerPressed event
-		private void RegisterPointerPressed(PointerEventHandler handler)
+		private protected override void OnNativeKeyDown(NSEvent evt)
 		{
-			LogRegisterPointerPressedNotImplemented();
-		}
+			Console.WriteLine($"[DOWN] Got key: {evt.KeyCode} | char: {evt.Characters} | char_no_modif: {evt.CharactersIgnoringModifiers}");
 
-		private void UnregisterPointerPressed(PointerEventHandler handler)
-		{
-			LogUnRegisterPointerPressedNotImplemented();
-		}
-#endregion
+			var args = new KeyRoutedEventArgs(this, System.VirtualKeyHelper.FromKeyCode(evt.KeyCode))
+			{
+				CanBubbleNatively = false // Only the first responder gets the event
+			};
 
-#region PointerReleased event
-		private void RegisterPointerReleased(PointerEventHandler handler)
-		{
-			LogRegisterPointerReleasedNotImplemented();
+			RaiseEvent(KeyDownEvent, args);
 		}
 
-		private void UnregisterPointerReleased(PointerEventHandler handler)
+		private protected override void OnNativeKeyUp(NSEvent evt)
 		{
-			LogUnRegisterPointerReleasedNotImplemented();
-		}
-#endregion
+			Console.WriteLine($"[UP] Got key: {evt.KeyCode} | char: {evt.Characters} | char_no_modif: {evt.CharactersIgnoringModifiers}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5545012)